### PR TITLE
chore(datasets): added remapping of headers in tabulate

### LINKED ISSUE #424: Add support for tabulate re-mapping of headers

### PR DESCRIPTION
**Pull request by jsam**
# Description

Display of headers on datasets list could be cleaner. Updated short_id to id and authors_csv to authors.
Fixes #424

### CODE REVIEW

File: renku/models/_tabulate.py [hunk 1]
```diff
@@ -32,8 +32,16 @@ def format_cell(cell, datetime_fmt=None):
 
 def tabulate(collection, headers, datetime_fmt='%Y-%m-%d %H:%M:%S', **kwargs):
     """Pretty-print a collection."""
+    if isinstance(headers, dict):
+        attrs = headers.keys()
+        # if mapping is not specified keep original
+        names = [
+            key if value is None else value for key, value in headers.items()
```
Comment (jirikuncar) [inline renku/models/_tabulate.py]: Do empty headers make sense? If not we can simplify to:

```suggestion
            value or key for key, value in headers.items()
```

Comment (jsam) [reply renku/models/_tabulate.py]: Imho, It makes a lot, due to partial mapping. I would actually rethink this entire solution which you initially proposed. But for now, I think it suffices.

File: renku/cli/dataset.py [hunk 2]
```diff
@@ -96,12 +95,24 @@ def dataset(ctx, client, datadir):
...
-            client.datasets.values(),
-            headers=['short_id', 'name', 'authors_csv', 'created'],
-        )
-        click.echo(table)
+    if ctx.invoked_subcommand is not None:
+        return
+
+    datasets = client.datasets.values()
+    if not datasets:
+        click.echo('No dataset records found.')
```
Comment (jirikuncar) [inline renku/cli/dataset.py]: This should probably go to `sys.stderr` (`err=True`).

```suggestion
        click.secho('No dataset records found.', fg='red', err=True)
```

File: renku/cli/dataset.py [hunk 3]
```diff
@@ -96,12 +95,24 @@ def dataset(ctx, client, datadir):
...
+        return
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
        ctx.exit(1)
        return
```

File: renku/cli/dataset.py [hunk 4]
```diff
@@ -96,12 +95,24 @@ def dataset(ctx, client, datadir):
...
+
+    table = tabulate(
+        datasets,
+        headers={
+            'short_id': 'id',
+            'name': None,
+            'created': None,
+            'authors_csv': 'authors'
+        }
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
        },
```

File: renku/cli/dataset.py [hunk 5]
```diff
@@ -96,12 +96,19 @@ def dataset(ctx, client, datadir):
...
-            client.datasets.values(),
-            headers=['short_id', 'name', 'authors_csv', 'created'],
-        )
-        click.echo(table)
+    if ctx.invoked_subcommand is not None:
+        return
+
+    output = tabulate(
+        client.datasets.values(),
+        headers={
```
Comment (jirikuncar) [inline renku/cli/dataset.py]: maybe we want to use `OrderedDict` for Python 3.5

File: tests/test_cli.py [hunk 6]
```diff
@@ -350,25 +350,33 @@ def test_datasets_import(data_file, data_repository, runner, project, client):
...
-def test_datasets_list_empty(runner, project):
+def test_datasets_list_empty(runner):
     """Test listing without datasets."""
     result = runner.invoke(cli.cli, ['dataset'])
     assert result.exit_code == 0
 
-    rows = result.output.split('\n')
-    assert len(rows) == 3
+    output = result.output.split('\n')
+    assert set(output.pop(0).split()) == {'AUTHORS', 'CREATED', 'ID', 'NAME'}
```
Comment (jirikuncar) [inline tests/test_cli.py]: with https://github.com/SwissDataScienceCenter/renku-python/pull/425/files#r259710131 you can keep using `list`

File: renku/cli/dataset.py [hunk 7]
```diff
@@ -96,12 +97,20 @@ def dataset(ctx, client, datadir):
...
-    if ctx.invoked_subcommand is None:
-        table = tabulate(
-            client.datasets.values(),
-            headers=['short_id', 'name', 'authors_csv', 'created'],
-        )
-        click.echo(table)
+    if ctx.invoked_subcommand is not None:
+        return
+
+    headers = OrderedDict()  # py35
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
    headers = OrderedDict((
        ('short_id', 'id'),
        ('name', None),
        ('created', None),
        ('authors_csv', 'authors'),
    ))
```

File: renku/cli/dataset.py [hunk 8]
```diff
@@ -96,12 +97,20 @@ def dataset(ctx, client, datadir):
...
+    headers['short_id'] = 'id'
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
```

File: renku/cli/dataset.py [hunk 9]
```diff
@@ -96,12 +97,20 @@ def dataset(ctx, client, datadir):
...
+    headers['name'] = None
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
```

File: renku/cli/dataset.py [hunk 10]
```diff
@@ -96,12 +97,20 @@ def dataset(ctx, client, datadir):
...
+    headers['created'] = None
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
```

File: renku/cli/dataset.py [hunk 11]
```diff
@@ -96,12 +97,20 @@ def dataset(ctx, client, datadir):
...
+    headers['authors_csv'] = 'authors'
```
Comment (jirikuncar) [inline renku/cli/dataset.py]:
```suggestion
```

**Pull request merged**